Updates the changelog and version number for the 0.5.0 release (#315)
XanaduAI/pennylane: .github/CHANGELOG.md

### Breaking changes

> * The method `Device.supported` was removed. The replacement methods are

Reviewer: ```suggestion
* The method `Device.supported` was removed, and replaced with the methods
```

    which will alert the user that an operation with `num_wires = All`
    is being incorrectly.

* The one-qubit rotations in `pennylane.plugins.default_qubit` no longer depend on Scipy's `expm`. Instead
  they are calculated with Euler's formula.
  [#292](https://github.com/XanaduAI/pennylane/pull/292)

* Creates an `ObservableReturnTypes` enumeration class containing `Sample`,
  `Variance` and `Expectation`. These new values can be assigned to the `return_type`
  attribute of an `Observable`.
  [#290](https://github.com/XanaduAI/pennylane/pull/290)

* Changed the signature of the `RandomLayer` and `RandomLayers` templates to have a fixed seed by default.
  [#258](https://github.com/XanaduAI/pennylane/pull/258)

* `setup.py` has been cleaned up, removing the non-working shebang,
  and removing unused imports.
  [#262](https://github.com/XanaduAI/pennylane/pull/262)

### Documentation

> * Fixes a typo in `QubitStateVector`.

Reviewer: ```suggestion
* Fixed a typo in `QubitStateVector`.
```

  [#295](https://github.com/XanaduAI/pennylane/pull/296)

> * Fixes a typo in the `default_gaussian.gaussian_state` function.

Reviewer: ```suggestion
* Fixed a typo in the `default_gaussian.gaussian_state` function.
```

  [#293](https://github.com/XanaduAI/pennylane/pull/293)

> * Fixes a typo in the gradient recipe within the `RX`, `RY`, `RZ`

Reviewer: ```suggestion
* Fixed a typo in the gradient recipe within the `RX`, `RY`, `RZ`
```

  operation docstrings.
  [#248](https://github.com/XanaduAI/pennylane/pull/248)

> * Fixes a broken link in the tutorial documentation, as a

Reviewer: ```suggestion
* Fixed a broken link in the tutorial documentation, as a
```
